core/server/worker/src/main/java/alluxio/worker/grpc/DoraWorkerClientServiceHandler.java

```diff
@@ -108,30 +118,67 @@ public StreamObserver<ReadRequest> readBlock(StreamObserver<ReadResponse> respon
   @Override
   public void getStatus(GetStatusPRequest request,
                         StreamObserver<GetStatusPResponse> responseObserver) {
+    FileInfo fi;
     try {
-      String ufsFilePath = request.getPath();
-      UfsFileStatus status = mUfsFileStatusCache.get(request.getPath());
-      GetStatusPResponse response = GetStatusPResponse.newBuilder()
-          .setFileInfo(
-              FileInfo.newBuilder()
-                  .setName(status.getName())
-                  .setPath(ufsFilePath)
-                  .setUfsPath(ufsFilePath)
-                  .setLength(status.getContentLength())
-                  .setBlockSizeBytes(status.getBlockSize())
-                  .setMode(status.getMode())
-                  .setFolder(status.isDirectory())
-                  .setLastModificationTimeMs(status.getLastModifiedTime())
-                  .setOwner(status.getOwner())
-                  .setGroup(status.getGroup())
-                  .setCompleted(true)
-                  .build()
-          ).build();
+      String alluxioFilePath = request.getPath();
+
+      String ufsFullPath = PathUtils.concatPath(mRootUFS, alluxioFilePath);
+      String fn = new AlluxioURI(alluxioFilePath).getName();
+
+      UfsFileStatus status = mUfsFileStatusCache.getIfPresent(ufsFullPath);
+      if (status == null) {
+        // The requested FileStatus is not present in memory cache.
+        // Let's try to query local persistent DoraMetaStore.
+        DoraMetaStore doraMetaStore = ((PagedDoraWorker) mWorker).getMetaStore();
+        Optional<DoraMeta.FileStatus> fs = doraMetaStore.getDoraMeta(ufsFullPath);
+        if (fs.isPresent()) {
+          // Found in persistent DoraMetaStore
+          fi = fs.get().getFileInfo();
+          String contentHash = UnderFileSystemUtils.approximateContentHash(fi.getLength(),
```

Comment:
I think it's fine for now but in the future I may need accurate contentHash for dstCP

Reply:
OK.